test/resource_test.rb

```diff
@@ -319,4 +398,21 @@ def destroy_resources
     end
     @resources = []
   end
+
+  def fork_workers(count:, resource: :testing, quota:, timeout: 0.1, sleep_seconds: 1)
+    (1..count).each do
+      fork do
+        begin
+          resource = Semian::Resource.new(resource.to_sym, quota: quota, timeout: timeout)
```

Comment:
Note that using create_resource doesn't work as expected from a threaded context, as the forked workers appear to try and run the "teardown" handler on their own / out of sync with the test runner.

This leads to a race that can cause forked processes to fail with EIDRM.

To avoid this, we spawn the workers ourselves and must use ensure on our own that they are killed / terminate. This is done with Process.waitall usually.